main.go

```diff
@@ -88,6 +107,31 @@ func main() {
 		os.Exit(1)
 	}
 
+	k8sClient, err := kubernetes.NewForConfig(mgr.GetConfig())
+	if err != nil {
+		setupLog.Error(err, "unable to create kubernets client")
+		os.Exit(1)
+	}
+
+	err = templates.InitializeSidecars(k8sClient)
+	if err != nil {
+		setupLog.Error(err, "unable to initialize sidecars")
+		os.Exit(1)
+	}
+
+	ctx := context.Background()
+	client, err := crclient.New(mgr.GetConfig(), crclient.Options{Scheme: scheme})
+	if err != nil {
+		setupLog.Error(err, "unable to create client")
+		os.Exit(1)
+	}
+
+	err = csi.CreateOrUpdateCSI(ctx, client, setupLog)
```

Comment:
Why are you doing this in `main` and not in its own controller? Shouldn't we have a controller that actually watches and reconciles the CSI sidecars rather than having to restart the operator to reconcile them?